core/widgets.py

```diff
@@ -0,0 +1,44 @@
+from typing import Any, Dict, Optional
+from django.forms import Widget
+
+class TagsEditor(Widget):
+    input_type = 'hidden'
+    template_name = 'core/widget_tag_editor.html'
+
+    @staticmethod
+    def __unpack_tag(tag):
+        return {'name': tag.name, 'color': tag.color}
+
+    def format_value(self, value: Any) -> Optional[str]:
+        if value is not None and not isinstance(value, str):
+            value = [self.__unpack_tag(tag) for tag in value]
+        return value
+    
+    def get_context(self, name: str, value: Any, attrs) -> Dict[str, Any]:
+        from . import models
```

Comment:
Any reason not to import this at the top? Just not a pattern I have used myself.

Reply:
Nope... just another case of muscle memory. Only needed for admin-panels I think(?) Whatever, not applicable in this case :D